pwa-devdocs/src/technologies/basic-concepts/client-side-caching/index.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Client-side caching
+---
+
+Client-server communication is slow and expensive.
+Performance is an important feature for any Progressive Web Application (PWA), so
+requests to the server should be minimized.
+
+Offline mode is also a required feature for a PWA.
+In offline mode, the application must be able to serve pages that have been recently viewed.
+
+These features are implemented with the help of a client-side cache.
+This local cache stores data from resources as they are fetched.
+Once a resource has been cached, the service worker may consult the cache on subsequent requests for that resource to boost performance.
+
+## Service Worker caching
+
+A [service worker][] is a JavaScript file that runs in a separate thread from the main execution thread in a web application.
+Service workers can intercept network requests and fetch cached data or store results from a network request into the cache.
+
+### Venia service worker
+
+Venia's service worker behavior is defined in the [`src/sw.js`][] file using Google's [Workbox][] library.
+
+You do not need to use Workbox to define service worker behavior, but
+Workbox makes this task easier by removing boilerplate code that is always used when working with service workers.
+
+Venia uses the following [caching strategies][] with its service worker:
+
+#### [Stale-while-revalidate][]
+
+The stale-while-revalidate strategy tells the service worker to use a cached response if it exists.
+A separate network request is made for that resource and the result is cached for future requests.
+
+This strategy is used when the most up to date version of a resource is not necessary for an application.
+
+| Route pattern                                     | Description          |
+| ------------------------------------------------- | -------------------- |
+| `/`                                               | The application root |
+| `/.\\.js$`                                        | JavaScript files     |
+| `/\/media\/catalog.*\.(?:png|gif|jpg|jpeg|svg)$/` | Catalog image files  |
```

Comment:
Not sure if Gatsby will treat this differently, but this table isn't showing up correctly:

![Screen Shot 2019-04-23 at 10 25 51 AM](https://user-images.githubusercontent.com/13182778/56594295-320fef80-65b2-11e9-8818-35e39d90db8f.png)

Reply:
Looks like GitHub's markdown processor isn't a fan of the `|` inside the regex.
Jekyll's markdown processor seems to be able to build it correctly though.